algebird-core/src/main/scala/com/twitter/algebird/Preparer.scala

```diff
@@ -0,0 +1,132 @@
+package com.twitter.algebird
+
+trait Preparer[A, T, +This[A, T] <: Preparer[A, T, This]] extends HasMonoidAggregate[A, T] {
```

Comment:
Yep, just wanted to get feedback on the API before I spent the time documenting it.